src/clients/HubPoolClient.ts

```diff
@@ -306,6 +332,45 @@ export class HubPoolClient extends BaseAbstractClient {
     return this.getTokenInfoForDeposit(deposit);
   }
 
+  /**
+   * @notice Return the bundle start block for the bundle containing the event with a given block number.
+   * @param eventBlock The event happened at this block on `eventChain`.
+   * @param eventChain The event happened on this chain.
+   * @param hubPoolLatestBlock Optional param that can be used to optimize the search time for which
+   * bundle contains the event
+   */
+  getBundleStartBlockContainingBlock(eventBlock: number, eventChain: number, hubPoolLatestBlock?: number): number {
+    // First find the latest executed bundle as of `hubPoolLatestBlock`.
+    const latestExecutedBundle = this.getNthFullyExecutedRootBundle(-1, hubPoolLatestBlock);
+
+    // If there is no latest executed bundle, then return the spoke's activation block. This means that there is no
+    // bundle before `hubPoolLatestBlock` containing the event block so the next bundle will start at
+    // the activation block and contain the event.
+    if (!latestExecutedBundle) {
```

Comment:
Should we be more explicit with these if statements? Not sure if it matters in practice, but technically `latestExecutedBundle = 0` will fail this conditional. We do this in other locations as well. We may want to employ the `isDefined(latestExecutedBundle)` type guard or just `latestExecutedBundle === undefined`

Reply:
`latestExecutedBundle` is a `ProposedRootBundle` object though

Reply:
Ah shoot - please disregard then. There's other locations in the code where we check the blocks which could be undefined or zero.

Reply:
I did add some `isDefined` though! good reminder